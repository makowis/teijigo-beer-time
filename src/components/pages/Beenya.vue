<template>
  <tbt-default>
    <div class="beenya">
      <tbt-h1>びーにゃの部屋</tbt-h1>
      <tbt-beenya-img :class="$style.icon" />
      <table :class="$style.beenyaTable">
        <tbody>
          <tr>
            <th>名前</th>
            <td>びーにゃ</td>
          </tr>
          <tr>
            <th>役割</th>
            <td>アイドル</td>
          </tr>
          <tr>
            <th>自己紹介</th>
            <td>
              にゃにゃにゃにゃにゃにゃにゃー！(定時後ビールタイムだー！！)
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </tbt-default>
</template>

<script lang="ts">
import Vue from 'vue';
import Component from 'vue-class-component';
import TbtDefault from '@/components/templates/TbtDefault.vue';
import TbtH1 from '@/components/atoms/TbtH1.vue';
import TbtBeenyaImg from '@/components/atoms/TbtBeenyaImg.vue';

@Component({
  components: {
    TbtDefault,
    TbtH1,
    TbtBeenyaImg,
  },
})
export default class Beenya extends Vue {}
</script>

<style module>
.icon {
  max-width: 90%;
  margin-bottom: 20px;
}

.beenyaTable {
  max-width: 90%;
  margin: auto;
  border: solid 1px black;
  border-collapse: separate;
}

.beenyaTable td,
.beenyaTable th {
  border: solid 1px black;
  padding: 5px 10px;
}
</style>
