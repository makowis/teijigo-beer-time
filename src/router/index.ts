import Vue from 'vue';
import Router from 'vue-router';
import Top from '@/components/pages/Top.vue';
import Disco from '@/components/pages/Disco.vue';
import Bbs from '@/components/pages/Bbs.vue';
import Member from '@/components/pages/Member.vue';
import Beenya from '@/components/pages/Beenya.vue';
import NotFound from '@/components/pages/NotFound.vue';

Vue.use(Router);

export default new Router({
  mode: 'history',
  routes: [
    {
      path: '/',
      name: 'Top',
      component: Top,
    },
    {
      path: '/disco',
      name: 'Disco',
      component: Disco,
    },
    {
      path: '/bbs',
      name: 'Bbs',
      component: Bbs,
    },
    {
      path: '/member',
      name: 'Member',
      component: Member,
    },
    {
      path: '/beenya',
      name: 'Beenya',
      component: Beenya,
    },
    {
      path: '*',
      name: 'NotFound',
      component: NotFound,
    },
  ],
  scrollBehavior(to, from, savedPosition) {
    if (savedPosition) {
      return savedPosition;
    }
    return { x: 0, y: 0 };
  },
});
