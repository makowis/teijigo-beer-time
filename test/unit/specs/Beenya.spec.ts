import Vue from 'vue';
import Beenya from '@/components/pages/Beenya.vue';
import router from '@/router/index';

describe('Beenya.vue', () => {
  const Constructor = Vue.extend(Beenya);
  const vm = new Constructor({ router }).$mount();

  it('should match the snapshot', () => {
    expect(vm.$el).toMatchSnapshot();
  });
});
