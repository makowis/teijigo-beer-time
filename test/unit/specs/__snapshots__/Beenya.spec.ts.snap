// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Beenya.vue should match the snapshot 1`] = `
<div
  id="app"
>
  <nav
    class="menuContainer"
    id="menu-container"
  >
    <ul
      class="menu"
    >
      <li
        class="menuItem"
      >
        <a
          class="router-link-exact-active router-link-active"
          href="/"
        >
          
        トップ
      
        </a>
      </li>
       
      <li
        class="menuItem"
      >
        <a
          class=""
          href="/disco"
        >
          
        DISCOGRAPHY
      
        </a>
      </li>
       
      <li
        class="menuItem"
      >
        <a
          class=""
          href="/member"
        >
          
        メンバー紹介
      
        </a>
      </li>
       
      <li
        class="menuItem"
      >
        <a
          class=""
          href="/bbs"
        >
          
        一言掲示板
      
        </a>
      </li>
       
      <li
        class="menuItem"
      >
        <a
          class=""
          href="/beenya"
        >
          
        びーにゃの部屋
      
        </a>
      </li>
       
      <li
        class="menuItem"
      >
        <a
          href="https://teijigo.booth.pm/items/1093403"
          target="_blank"
        >
          
        通販(BOOTH)
      
        </a>
      </li>
    </ul>
  </nav>
   
  <div
    class="mainContainer"
    id="main-container"
  >
    <a
      class="router-link-exact-active router-link-active"
      href="/"
    >
      <img
        class="logo"
        src="@/assets/logo_1.png"
      />
    </a>
     
    <div
      class="beenya"
    >
      <h1
        class="h1"
      >
        びーにゃの部屋
      </h1>
       
      <img
        alt="びーにゃのイラスト"
        class="icon"
        src="@/assets/beenya.png"
      />
       
      <table
        class="beenyaTable"
      >
        <tbody>
          <tr>
            <th>
              名前
            </th>
             
            <td>
              びーにゃ
            </td>
          </tr>
           
          <tr>
            <th>
              役割
            </th>
             
            <td>
              アイドル
            </td>
          </tr>
           
          <tr>
            <th>
              自己紹介
            </th>
             
            <td>
              
            にゃにゃにゃにゃにゃにゃにゃー！(定時後ビールタイムだー！！)
          
            </td>
          </tr>
        </tbody>
      </table>
    </div>
     
    <footer
      class="footer"
    >
      <p>
        ©定時後ビールタイム
      </p>
    </footer>
  </div>
</div>
`;
